src/data/finsupp.lean

```diff
@@ -125,6 +133,29 @@ if_neg hb
 lemma support_single_subset : (single a b).support ⊆ {a} :=
 show ite _ _ _ ⊆ _, by split_ifs; [exact empty_subset _, exact subset.refl _]
 
+lemma injective_single (a : α) : function.injective (single a : β → α →₀ β) :=
+assume b₁ b₂ eq,
+have (single a b₁ : α →₀ β) a = (single a b₂ : α →₀ β) a, by rw eq,
+by rwa [single_eq_same, single_eq_same] at this
+
+lemma single_eq_single_iff (a₁ a₂ : α) (b₁ b₂ : β) :
+  single a₁ b₁ = single a₂ b₂ ↔ ((a₁ = a₂ ∧ b₁ = b₂) ∨ (b₁ = 0 ∧ b₂ = 0)):=
```

Comment:
Doesn't `injective_single` follow easily from this proof?

Reply:
The proof uses `injective_single`